growthbook.go

```diff
@@ -97,136 +346,277 @@ func (fr *FeatureResult) GetValueWithDefault(def FeatureValue) FeatureValue {
 	return fr.Value
 }
 
+// IsOn determines whether a feature is on.
+func (gb *GrowthBook) IsOn(key string) bool {
+	gb.inner.RLock()
+	defer gb.inner.RUnlock()
+
+	return gb.EvalFeature(key).On
+}
+
+// IsOff determines whether a feature is off.
+func (gb *GrowthBook) IsOff(key string) bool {
+	gb.inner.RLock()
+	defer gb.inner.RUnlock()
+
+	return gb.EvalFeature(key).Off
+}
+
+// GetFeatureValue returns the result for a feature identified by a
+// string feature key, with an explicit default.
+func (gb *GrowthBook) GetFeatureValue(key string, defaultValue interface{}) interface{} {
+	gb.inner.RLock()
+	defer gb.inner.RUnlock()
+
+	featureValue := gb.EvalFeature(key).Value
+	if featureValue != nil {
+		return featureValue
+	}
+	return defaultValue
+}
+
 // Feature returns the result for a feature identified by a string
-// feature key.
+// feature key. (DEPRECATED: Use EvalFeature instead.)
```

Comment:
If you use the `// Deprecated: ` prefix in the documentation comment, this will be picked up by editors. In the below screenshots I just marked any function as deprecated and it is supported in editors that support this level of intellisense.

<img width="852" alt="image" src="https://github.com/growthbook/growthbook-golang/assets/113377031/13134746-4d96-496a-af1d-9dad023868d7">
<img width="436" alt="image" src="https://github.com/growthbook/growthbook-golang/assets/113377031/3227f7a3-abec-47a5-88ce-9ec13f37aa6e">

You will see in the below screenshot, the syntax chosen will not properly indicate the function is deprecated.

<img width="1515" alt="image" src="https://github.com/growthbook/growthbook-golang/assets/113377031/2aefe7ca-175d-418d-a46a-c0a137489abd">

You can read more about this in the official Go lang docs: https://tip.golang.org/ref/mod#go-mod-file-module-deprecation

Reply:
Thanks! I should have known that. I've seen it, but I didn't realize was something godoc and the LSP server understood.